### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by ptrkcsk**
This PR:

- Updates all Composer dependencies
- Adds local Composer installation via the Makefile so project users don't need to worry about installing Composer

I believe this will be my first contribution since we transferred this repo from @ptrkcsk to @wp-cli

@danielbachhuber Looks like you're no longer a WP-CLI maintainer, but wanted to confirm

@schlessera Looks like you're still a WP-CLI maintainer. Would you like to review this PR?

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -18,21 +18,25 @@
     }
   ],
   "require": {
-    "php": ">=7",
+    "php": "7.4 - 8.1",
```
Comment (danielbachhuber) [inline composer.json:21]: Should this be `>= 7.4 || >= 8`?

**Review by danielbachhuber — COMMENTED** [review]
Seems fine on my end, @ptrkcsk! Are there any specific pieces you have a question about?

File: composer.json (hunk 1 above)
Comment (ptrkcsk) [reply composer.json:21]: I find the hyphenated range easier to read, and I like to intentionally smoke test before bumping the upper bound when the next version of PHP is released.

I'm open to changing it though. I haven't worked with WP or PHP for a while.

**Comment by ptrkcsk** [comment]
Thanks for the review, @danielbachhuber !

> Are there any specific pieces you have a question about?

No, just wanted to dust off the project a little

File: composer.json (hunk 1 above)
Comment (danielbachhuber) [reply composer.json:21]: > I find the hyphenated range easier to read, and I like to intentionally smoke test before bumping the upper bound when the next version of PHP is released.

Cool, works for me 😊 I don't have a strong opinion about it.

**Review by danielbachhuber — APPROVED** [review]

**Pull request merged**